### DISCUSSION TITLE
Vertex color

### DISCUSSION THREAD

**Discussion by lauris47** [SteamDatabase/ValveResourceFormat#679] [General]
Hi,
Does Source 2 Viewer supports decompiling of Vertex point coloring? 
I am asking, because it looks different than what it was before in CSGO. 
So either CS2 is using a different technique, or Source 2 Viewer exports this wrongly. 
I know it is new engine, so it actually might be a different technique. 

I can look at vertex paint, and in CSGO 4-way-blend (or 2 texture blending) was done mostly on alpha value of the vertex color. 

In cs2, if I use alpha, the result will be wrong.

Any ideas, or maybe a confirmation?

**Comment by kristiker** [comment]
Vertex blend paint should be in texcoord3 or texcoord4 in blender 4.0

**Reply by lauris47** [reply]
@kristiker Thanks.
Yes, I noticed there is some painting values in texcoord4, but is it correctly implemented?
I am yet to find how to use it to get to what i see in CS2.

**Reply by kristiker** [reply]
Depens on the shader used. Mirage materials are same as source1. Inferno's are different.

Though worth noting Source 2 Viewer does not export the materials in blendable form. No 2nd layer or blendmask.

**Reply by kristiker** [reply]
Inferno materials most likely use some some type of the implementation found in this channel https://www.youtube.com/@simonfuchs1812/videos
